document-store/src/main/java/org/hypertrace/core/documentstore/model/config/TypesafeConfigDatastoreConfigExtractor.java

```diff
@@ -78,80 +87,80 @@ public TypesafeConfigDatastoreConfigExtractor keysForEndpoints(
   }
 
   public TypesafeConfigDatastoreConfigExtractor hostKey(@NonNull final String key) {
-    if (config.hasPath(key)) {
+    if (config.hasPath(key) && STRING.equals(config.getValue(key).valueType())) {
       endpointBuilder.host(config.getString(key));
     }
     return this;
   }
 
   public TypesafeConfigDatastoreConfigExtractor portKey(@NonNull final String key) {
-    if (config.hasPath(key)) {
+    if (config.hasPath(key) && NUMBER.equals(config.getValue(key).valueType())) {
```

Comment:
What would happen if the port value is `STRING`? would that is not parsed?
Most of the places for config, we already know the types as config is driven manually.

Reply:
If the port is a string, it throws an exception without an explicit type check. Hence, added the safety type-check everywhere in addition to the existence check.

Reply:
So far, we didn't have type checking in the code, so if type checked is enforced, it should throw the exception if desired type is not matched.

And if there are multiple way to provides same configuration, then we should throw an exception if none of the valid configuration found.  E.g if say for connection, it can be configured either using
- connectionString
- host + port
So, if both are invalids, we should throw the exception. It's better to exit early with an error rather than discovering the issue while executing the query that connection error, etc.

Reply:
Makes sense. Updated the PR to reactively consume "host" and "port" for older configuration maintaining backward compatibility and reverted the change for type-checking.